server/api/src/rsDataObjPhymv.cpp

```diff
@@ -17,378 +17,227 @@
+        addKeyVal(&source_data_obj_inp.condInput, RESC_HIER_STR_KW, hier.c_str());
+        return source_data_obj_inp;
+    }
+
+    int close_replica(
+        rsComm_t* rsComm,
+        const int _inx,
+        const int _status) {
+        openedDataObjInp_t dataObjCloseInp{};
+        dataObjCloseInp.l1descInx = _inx;
+        L1desc[dataObjCloseInp.l1descInx].oprStatus = _status;
+        char* pdmo_kw = getValByKey(&L1desc[_inx].dataObjInp->condInput, IN_PDMO_KW);
+        if (pdmo_kw) {
+            addKeyVal(&dataObjCloseInp.condInput, IN_PDMO_KW, pdmo_kw);
+        }
+        const int status = rsDataObjClose( rsComm, &dataObjCloseInp);
+        if (status < 0) {
+            rodsLog(LOG_ERROR, "[%s] - rsDataObjClose failed with [%d]", __FUNCTION__, status);
+        }
+        clearKeyVal( &dataObjCloseInp.condInput );
         return status;
     }
 
-    char* dest_resc = getValByKey( &dataObjInp->condInput, DEST_RESC_NAME_KW );
-    if ( dest_resc ) {
-        std::string dest_hier;
-        irods::error ret = resolve_hierarchy_for_resc_from_cond_input(
-                               rsComm,
-                               dest_resc,
-                               dest_hier );
-        if( !ret.ok() ) {
-            irods::log( PASS( ret ) );
-            return ret.code();
+    int open_source_replica(
+        rsComm_t* rsComm,
+        dataObjInp_t& source_data_obj_inp) {
+        source_data_obj_inp.oprType = PHYMV_SRC;
+        source_data_obj_inp.openFlags = O_RDWR;
+        int source_l1descInx = rsDataObjOpen(rsComm, &source_data_obj_inp);
+        if (source_l1descInx < 0) {
+            return source_l1descInx;
         }
-        addKeyVal(
-            &dataObjInp->condInput,
-            DEST_RESC_HIER_STR_KW,
-            dest_hier.c_str() );
-    }
-
-    // =-=-=-=-=-=-=-
-    // determine hierarchy string
-	char* dest_hier_kw = getValByKey( &dataObjInp->condInput, DEST_RESC_HIER_STR_KW );
-    std::string dest_hier;
-    if ( NULL == dest_hier_kw || 0 == strlen(dest_hier_kw) ) {
-        irods::error ret = irods::resolve_resource_hierarchy(
-			irods::CREATE_OPERATION,
-			rsComm,
-			dataObjInp,
-			dest_hier );
-        if ( !ret.ok() ) {
-            std::stringstream msg;
-            msg << __FUNCTION__;
-            msg << " :: failed in irods::resolve_resource_hierarchy for [";
-            msg << dataObjInp->objPath << "]";
-            irods::log( PASSMSG( msg.str(), ret ) );
-            return ret.code();
+        return source_l1descInx;
+    }
+
+    int open_destination_replica(
+        rsComm_t* rsComm,
+        dataObjInp_t& destination_data_obj_inp) {
+        addKeyVal(&destination_data_obj_inp.condInput, REG_REPL_KW, "");
+        addKeyVal(&destination_data_obj_inp.condInput, FORCE_FLAG_KW, "");
+        destination_data_obj_inp.oprType = REPLICATE_DEST;
+        destination_data_obj_inp.oprType = PHYMV_DEST;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'destination_data_obj_inp.oprType' is reassigned a value before the old one has been used.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=4331469)